test/tube_test.c

```diff
@@ -268,3 +268,39 @@ CTEST2(tube, print_tubes)
     ASSERT_TRUE(tube_manager_add(data->mgr, t, &data->err));
     tube_manager_print_tubes(data->mgr);
 }
+
+static int _mock_tube_walker(void *data,
+                             const spud_tube_id *tube_id,
+                             tube *t)
+{
+    UNUSED_PARAM(tube_id);
+    UNUSED_PARAM(t);
+    int *num_tubes = (int *) data;
+    (*num_tubes)++;
+    return 1;
```

Comment:
I tried destroying the tube but am getting double free errors. I looked deeper into the code, and it looks like tubes are automatically destroyed when their containing tube manager is destroyed. The `tube_manager_add` function passes the cleaner function `clean_tube` to `ls_htable_put`. I believe `clean_tube` is being called on `tube_manager_destroy`=>`ls_htable_destroy`, correct?